pkg/apis/autoscaling/validation/validation.go

```diff
@@ -103,14 +103,28 @@ func ValidateHorizontalPodAutoscalerStatusUpdate(newAutoscaler, oldAutoscaler *a
 	return allErrs
 }
 
-func validateMetrics(metrics []autoscaling.MetricSpec, fldPath *field.Path) field.ErrorList {
+func validateMetrics(metrics []autoscaling.MetricSpec, fldPath *field.Path, minReplicas *int32) field.ErrorList {
 	allErrs := field.ErrorList{}
+	hasObjectMetrics := false
+	hasExternalMetrics := false
 
 	for i, metricSpec := range metrics {
 		idxPath := fldPath.Index(i)
 		if targetErrs := validateMetricSpec(metricSpec, idxPath); len(targetErrs) > 0 {
 			allErrs = append(allErrs, targetErrs...)
 		}
+		if metricSpec.Type == autoscaling.ObjectMetricSourceType {
+			hasObjectMetrics = true
+		}
+		if metricSpec.Type == autoscaling.ExternalMetricSourceType {
+			hasExternalMetrics = true
+		}
+	}
+
+	if minReplicas != nil && *minReplicas == 0 {
```

Comment:
Should be if it's nil or zero?

Reply:
We need at least one object or external metric configured if minReplicas is set to zero. Not nil check is required to avoid nil pointer dereference.

Reply:
Officially it defaults to 1, but the defaulting code should have already run by the time this function is invoked. So, it's not clear what nil means at this point, I was assuming 0...

Reply:
@lavalamp minReplicas is a pointer to int32.

I've done the same nil protective check as [here](https://github.com/kubernetes/kubernetes/pull/74526/files/1e9ec7b634181f4f9220d0cc10c9381b3e410986#diff-4364e3f754b12e39f60eea33105f6c7eR47)

It looks robust and doesn't require to know about external defaulting code.

Reply:
this check looks fine. practically, it will not be nil since defaulting runs before we get here, but the nil check makes it clearly safe